pkgs/test_core/lib/src/runner.dart

```diff
@@ -324,6 +326,17 @@ class Runner {
             // The absolute path as it will appear in stack traces.
             var absoluteSuitePath = File(path).absolute.uri.toFilePath();
 
+            // First check if we're a match for the overridden location.
+            if (location != null) {
+              if ((line == null || location.line == line) &&
+                  (col == null || location.column == col) &&
+                  location.uri.isScheme('file') &&
+                  location.uri.toFilePath() == absoluteSuitePath) {
```

Comment:
I've assumed we're ok to only handle file URIs here. The code below for stack traces does handle another case (and mentions `--filesystem-scheme`) but it's not clear to me under what circumstances that would be used. If we need to support some equivalent here, I'll need to better understand what that is.

Reply:
We do have other URIs when tests are pre-compiled.

Specifically `org-dartlang-app:///` will be used among others.

Reply:
Specifically these are used to obfuscate the file system - so the paths are relative to some other root which isn't machine specific (its some path under the repository, usually the root of the package or repository)

Reply:
I've extracted the URI check into a new helper `matchesUri` that this now calls (as does the original `matchLineAndCol`).

Reply:
Although I also see that this code won't work right on Windows, since `absoluteSuitePath` is a real path (`.toFilePath()`) and is compared to a URI path (`absoluteSuitePath.endsWith(uri.path)`) that would have the wrong slashes on Windows. I'll tweak this to always use "URI paths".